src/core/public/application/ui/app_container.tsx

```diff
@@ -107,12 +108,14 @@ export const AppContainer: FunctionComponent<Props> = ({
   return (
     <Fragment>
       {appNotFound && <AppNotFound />}
-      {showSpinner && (
-        <div className="appContainer__loading">
-          <EuiLoadingElastic aria-label="Loading application" size="xxl" />
-        </div>
+      {showSpinner && !appNotFound && (
+        <EuiLoadingElastic
+          className="appContainer__loading"
+          aria-label="Loading application"
+          size="xxl"
+        />
       )}
-      <div key={appId} ref={elementRef} />
+      <div className={APP_WRAPPER_CLASS} key={appId} ref={elementRef} />
```

Comment:
Sorry, I can't help myself. A quick a11y win we could grab here.

```suggestion
      <div className={APP_WRAPPER_CLASS} key={appId} ref={elementRef} aria-busy={showSpinner} />
```